plugins/router/arris-sb6183

```diff
@@ -0,0 +1,366 @@
+        for i in range(1, DOWNCOUNT + 1):
+            print("down_{0}.label Ch {1}".format(i, i))
+            print("down_{0}.type GAUGE".format(i))
+            print("down_{0}.draw LINE1".format(i))
+
+        for i in range(1, DOWNCOUNT + 1):
+            name = "down_{0}".format(i)
+            print("\nmultigraph arris_snr.{0}".format(name))
+            print("graph_title SNR on Channel {0} (dB)".format(i))
+            print("graph_vlabel SNR (dB)")
+            print("graph_category network")
+            print("snr.label dB")
+            print("snr.type GAUGE")
+            print("snr.draw LINE1")
+
+        # ERRORS
+        print(
+            """
+multigraph arris_error
+graph_title Arris Channel Errors
+graph_category network
+graph_args --base 1000
+graph_vlabel errors/sec
+graph_category network
+corr.label Corrected
+corr.type DERIVE
+corr.min 0
+uncr.label Uncorrectable
+uncr.type DERIVE
+uncr.min 0
+uncr.warning 1"""
+        )
+
+        for i in range(1, DOWNCOUNT + 1):
+            name = "down_{0}".format(i)
+            print("\nmultigraph arris_error.{0}".format(name))
+            print("graph_title Channel {0} Errors".format(i))
+            print("graph_args --base 1000")
+            print("graph_vlabel errors/sec")
+            print("graph_category network")
+            print("corr.label Correctable")
+            print("corr.type DERIVE")
+            print("corr.min 0")
+            print("uncr.label Uncorrectable")
+            print("uncr.type DERIVE")
+            print("uncr.min 0")
+            print("uncr.warning 1")
+
+        sys.exit(0)
+
+    if sys.argv[1] == "autoconfig":
+
+        def check(url):
+            from lxml import html
+
+            resp = request.urlopen(url)
+            html.fromstring("<html>")
+            return resp
+
+        try:
+            resp = check(STATUS_URL)
+        except ImportError:
+            print("no (missing lxml module)")
+        except OSError:
+            print("no (no router)")
+        else:
+            if resp.status == 200:
+                print("yes")
+            else:
+                print("no (Bad status code: %d)" % resp.status_code)
+        sys.exit(0)
+
+rxblank = re.compile(r"[\x00\n\r\t ]+", re.MULTILINE)
+rxcomment = re.compile(r"<!--.*?-->")
+rxscript = re.compile(r"<script.*?</script>", re.MULTILINE)
+
+
+def process_url(url):
+    """
+    Extract simpleTables from page at URL
+    """
+    from lxml import html
+
+    try:
+        resp = request.urlopen(url)
+    except OSError:
+        print("failed to contact router", file=sys.stderr)
+        return []
+    if resp.status != 200:
+        print(
+            "failed to get status page %d: %s" % (resp.status, resp.reason),
+            file=sys.stderr,
+        )
+        return []
+    data = rxscript.sub(
+        "",
+        rxcomment.sub(
+            "",
+            rxblank.sub(
+                " ", "".join(map(lambda x: x.decode("utf-8"), resp.readlines()))
```

Comment:
Just a style recommendation - maybe a generator comprehension would be easier to read:
```python
"".join(line.decode() for line in resp.readlines())
```

Or maybe just use `resp.read().decode()`?